src/twisted/conch/ssh/userauth.py

```diff
@@ -124,37 +132,40 @@ def timeoutAuthentication(self):
             transport.DISCONNECT_NO_MORE_AUTH_METHODS_AVAILABLE, b"you took too long"
         )
 
-    def tryAuth(self, kind, user, data):
+    def tryAuth(
+        self, kind: bytes, user: bytes, data: bytes
+    ) -> Deferred[_ConchPortalTuple]:
         """
         Try to authenticate the user with the given method.  Dispatches to a
         auth_* method.
 
         @param kind: the authentication method to try.
-        @type kind: L{bytes}
+
         @param user: the username the client is authenticating with.
-        @type user: L{bytes}
+
         @param data: authentication specific data sent by the client.
-        @type data: L{bytes}
+
         @return: A Deferred called back if the method succeeded, or erred back
             if it failed.
-        @rtype: C{defer.Deferred}
         """
         self._log.debug("{user!r} trying auth {kind!r}", user=user, kind=kind)
         if kind not in self.supportedAuthentications:
             return defer.fail(error.ConchError("unsupported authentication, failing"))
-        kind = nativeString(kind.replace(b"-", b"_"))
-        f = getattr(self, f"auth_{kind}", None)
-        if f:
+        strkind = nativeString(kind.replace(b"-", b"_"))
```

Comment:
Minor comment

I think that we can stop using `nativeString` for the refactored code.

```suggestion
        strkind = kind.replace(b"-", b"_").decode("ascii")
```